api/v1/server/handlers/monitoring/probe.go

```diff
@@ -0,0 +1,317 @@
+		}
+
+		return gen.MonitoringPostRunProbe403JSONResponse{}, err
+	}
+
+	token, err := getBearerTokenFromRequest(ctx.Request())
+
+	if err != nil {
+		return nil, err
+	}
+
+	events := make(chan string, 50)
+	errorChan := make(chan error, 50)
+	interrupt := cmdutils.InterruptChan()
+
+	cancellableContext, cancel := context.WithTimeout(ctx.Request().Context(), m.probeTimeout)
+
+	defer cancel()
+	var cleanup func() error
+
+	cf := clientconfig.ClientConfigFile{
+		Token:     token,
+		TenantId:  tenant.ID,
+		Namespace: randomNamespace(),
+		TLS: clientconfig.ClientTLSConfigFile{
+			Base: shared.TLSConfigFile{
+				TLSRootCAFile: os.Getenv("SERVER_TLS_ROOT_CA_FILE"),
+			},
+		},
+	}
+	workflowName := "probe-workflow"
+	var wfrId *string
+	go func() {
+		wfrId, cleanup, err = m.run(cancellableContext, cf, workflowName, events, errorChan)
+		if err != nil {
+			errorChan <- err
+		}
+
+		// check if the context was cancelled in the meantime
+
+		if cancellableContext.Err() != nil {
+			cleanupErr := cleanup()
+			if cleanupErr != nil {
+				m.l.Error().Msgf("error cleaning up probe worker: %s", cleanupErr)
+			}
+		}
+	}()
+
+	defer func() {
+		var cleanupErr error
+		if cleanup != nil {
+			cleanupErr = cleanup()
+		} else {
+			m.l.Warn().Msg("cleanup function was never set for probe")
+		}
+		if cleanupErr != nil {
+			m.l.Error().Msgf("error cleaning up probe worker: %s", err)
+		}
+	}()
+
+	defer func() {
+
+		time.Sleep(250 * time.Millisecond)
+
+		if wfrId != nil && *wfrId != "" {
+
+			wrfRow, err := m.config.APIRepository.WorkflowRun().GetWorkflowRunById(context.Background(), cf.TenantId, *wfrId)
+
+			if err != nil {
+				m.l.Error().Msgf("error getting workflow run: %s", err)
+
+			}
+
+			workflowId := sqlchelpers.UUIDToStr(wrfRow.Workflow.ID)
+
+			_, err = m.config.APIRepository.Workflow().DeleteWorkflow(cf.TenantId, workflowId)
+
+			m.l.Info().Msgf("deleted workflow %s", workflowId)
+			if err != nil {
+				m.l.Error().Msgf("error deleting workflow: %s", err)
+			}
+		} else {
+			m.l.Warn().Msg("workflow run id was never set")
+		}
+
+	}()
+
+	messages := []string{"This is a stream event for step-one", "This is a stream event for step-two"}
+	messageIndex := 0
+
+	for {
+
+		select {
+		case e := <-events:
+			if e != messages[messageIndex] {
+				return nil, fmt.Errorf("expected message %s, got %s", messages[messageIndex], e)
+			}
+
+			messageIndex++
+			if messageIndex == len(messages) {
+				m.l.Debug().Msg("probe completed successfully")
+				return gen.MonitoringPostRunProbe200JSONResponse{}, nil
+			}
+
+		case <-cancellableContext.Done():
+
+			if cancellableContext.Err() == context.DeadlineExceeded {
+				m.l.Error().Msg("timed out waiting for probe to complete")
+				return nil, fmt.Errorf("timed out waiting for probe to complete")
+			} else {
+				m.l.Debug().Msg("probe completed successfully")
+				return gen.MonitoringPostRunProbe200JSONResponse{}, nil
```

Comment:
This case seems be returning a 200 response when the context gets cancelled without us necessarily checking any correctness conditions -- what's the thinking here?